File: VotingSystem.Shared/Models/VoteResponseDto.cs
namespace VotingSystem.Shared.Models;

public record VoteResponseDto
{
    public required int Id { get; init; }
    
    public required UserResponseDto User { get; init; }
    
    public required string Question { get; init; }
    
    public required string[] Options { get; init; }

    public required DateTime Start { get; init; }
    
    public required DateTime End { get; init; }
    
    public required ICollection<AnonymousVoteResponseDto> AnonymousVotes { get; set; }
    
    public required ICollection<VoteParticipationResponseDto> VoteParticipations { get; set; }
    
    public DateTime CreatedAt { get; init; }
    
    public DateTime UpdatedAt { get; init; }
}

File: votingsystem.react/src/api/models/VoteResponseDto.ts
import {UserResponseDto} from "./UserResponseDto.ts";
import { AnonymousVoteResponseDto } from "./AnonymousVoteResponseDto";
import { VoteParticipationResponseDto } from "./VoteParticipationResponseDto";

export interface VoteResponseDto {
    id: number;
    user: UserResponseDto;
    question: string;
    options: string[];
    start: Date;
    end: Date;
    anonymousVotes: AnonymousVoteResponseDto[];
    voteParticipation: VoteParticipationResponseDto[];
    createdAt: Date;
    updatedAt: Date;
}
